### ISSUE TITLE
google-auth-library-nodejs@6 dependency breaks node 8 deploys

### ISSUE THREAD

**Issue by alexlouden** [firebase/firebase-tools#2284]
### [REQUIRED] Environment info

**firebase-tools:** 8.0.3

**Platform:** ubuntu

### [REQUIRED] Test case

Deploys have started failing with node 8 with the latest version of firebase-tools.

### [REQUIRED] Steps to reproduce

```bash
yarn add firebase-tools@8.0.3

firebase deploy
...
⚠  functions[myfunction(us-central1)]: Deployment error. 
Build failed: `yarn_install` had stderr output: 
error google-auth-library@6.0.0: The engine "node" is incompatible with this module. Expected version ">=10". 
error Found incompatible module
```

### [REQUIRED] Expected behavior

The ability to deploy node 8 functions

### [REQUIRED] Actual behavior

This seems to be coming from the latest version of `@grpc/grpc-js`, which depends on `google-auth-library@6.0.0`, which is [now](https://github.com/googleapis/google-auth-library-nodejs/releases/tag/v6.0.0) node 10 only.

See `yarn why`:

```bash
yarn why google-auth-library
...
=> Found "firebase-tools#@grpc/grpc-js#google-auth-library@6.0.0"
info This module exists because "firebase-tools#@google-cloud#pubsub#google-gax#@grpc#grpc-js" depends on it.
```

**Comment by aravindvnair99** [comment]
@alexlouden [Node.js 8 is deprecated. Move along to Node.js 10](https://firebase.google.com/docs/functions/manage-functions#set_runtime_options).

**Comment by alexlouden** [comment]
> @alexlouden [Node.js 8 is deprecated. Move along to Node.js 10](https://firebase.google.com/docs/functions/manage-functions#set_runtime_options).

I'm aware, however the end date for node 8 support is February next year. 

> Starting February 15, 2021, we'll no longer support new deploys and updates of Node.js 8 functions. Then, starting March 15, 2021, we'll no longer support executions of Node.js 8 functions.

**Comment by aravindvnair99** [comment]
> > @alexlouden [Node.js 8 is deprecated. Move along to Node.js 10](https://firebase.google.com/docs/functions/manage-functions#set_runtime_options).
> 
> I'm aware, however the end date for node 8 support is February next year.
> 
> > Starting February 15, 2021, we'll no longer support new deploys and updates of Node.js 8 functions. Then, starting March 15, 2021, we'll no longer support executions of Node.js 8 functions.

@alexlouden Yes, but you installed v6.0.0 which is not supported on Node.js 8. You need to install v5.10.1 otherwise.

**Comment by alexlouden** [comment]
No, I installed `firebase-tools`, and one of the dependencies installed it v6, which breaks node 8 support - that's the problem.

**Comment by samtstern** [comment]
@alexlouden thank you for filing this!  You're right, many Google Node libraries recently dropped support for Node 8.  However we (the Firebase CLI) intend to keep Node 8 support for a little longer so we need to figure out how to get this dependency fixed.

**Comment by samtstern** [comment]
@alexlouden does this also happen to you on the latest version of `firebase-tools` (`8.4.0`)?

**Comment by samtstern** [comment]
I actually can't see anything in our dep chain that leads to `google-auth-library` 6+:

```
$ npm ls google-auth-library
firebase-tools@8.4.0 /Users/samstern/Projects/firebase-tools
├─┬ @google-cloud/pubsub@1.7.0
│ ├── google-auth-library@5.5.0  deduped
│ └─┬ google-gax@1.15.2
│   └── google-auth-library@5.5.0  deduped
├─┬ firebase-admin@8.10.0
│ ├─┬ @google-cloud/firestore@3.7.1
│ │ └─┬ google-gax@1.15.1
│ │   └── google-auth-library@5.10.1 
│ └─┬ @google-cloud/storage@4.6.0
│   ├─┬ @google-cloud/common@2.4.0
│   │ └── google-auth-library@5.10.1 
│   └─┬ gcs-resumable-upload@2.3.3
│     └── google-auth-library@5.10.1 
├── google-auth-library@5.5.0 
└─┬ google-gax@1.12.0
  └── google-auth-library@5.8.0 
```

**Comment by alexlouden** [comment]
Thanks for that @samtstern!

This started happening on our CI environment 4 days ago. This is the current version:

```bash
$ npm ls google-auth-library
├─┬ firebase-tools@8.0.3
│ ├─┬ @google-cloud/pubsub@1.7.3
│ │ ├── google-auth-library@5.7.0  deduped
│ │ └─┬ google-gax@1.15.3
│ │   ├─┬ @grpc/grpc-js@1.0.4
│ │   │ └── google-auth-library@6.0.0       <---
```

And the latest version (installed with yarn):

```bash
$ yarn add firebase-tools@8.4.0
...
$ npm ls google-auth-library
├─┬ firebase-tools@8.4.0
│ ├─┬ @google-cloud/pubsub@1.7.3
│ │ ├── google-auth-library@5.7.0  deduped
│ │ └─┬ google-gax@1.15.3
│ │   ├─┬ @grpc/grpc-js@1.0.4
│ │   │ └── google-auth-library@6.0.0       <---
```

If I install it clean using `yarn add` it installs v6:
```
$ rm -r node_modules yarn.lock package.json
$ yarn add firebase-tools@8.4.0
yarn add v1.21.1
info No lockfile found.
[1/4] Resolving packages...
...
[2/4] Fetching packages...
info fsevents@2.1.3: The platform "linux" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 351 new dependencies.
...

$ npm ls google-auth-library
└─┬ firebase-tools@8.4.0
  ├─┬ @google-cloud/pubsub@1.7.3
  │ ├── google-auth-library@5.10.1  deduped
  │ └─┬ google-gax@1.15.3
  │   ├─┬ @grpc/grpc-js@1.0.4
  │   │ └── google-auth-library@6.0.0       <---
  │   └── google-auth-library@5.10.1  deduped
  ├── google-auth-library@5.10.1
  └─┬ google-gax@1.12.0
    └── google-auth-library@5.10.1  deduped
```

The CI env:
```bash
$ node -v
v8.17.0
$ yarn -v
1.21.1
$ uname -a
Linux xxx 4.15.0-1052-aws #54-Ubuntu SMP Tue Oct 1 15:43:26 UTC 2019 x86_64 GNU/Linux
```

However it installs the older version with `npm`, with a notsup warning :

```bash
npm install firebase-tools
...
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN notsup Unsupported engine for google-auth-library@6.0.0: wanted: {"node":">=10"} (current: {"node":"8.17.0","npm":"6.13.4"})
npm WARN notsup Not compatible with your version of node/npm: google-auth-library@6.0.0
npm WARN notsup Unsupported engine for gtoken@5.0.1: wanted: {"node":">=10"} (current: {"node":"8.17.0","npm":"6.13.4"})
npm WARN notsup Not compatible with your version of node/npm: gtoken@5.0.1
npm WARN notsup Unsupported engine for gcp-metadata@4.1.0: wanted: {"node":">=10"} (current: {"node":"8.17.0","npm":"6.13.4"})
npm WARN notsup Not compatible with your version of node/npm: gcp-metadata@4.1.0
npm WARN notsup Unsupported engine for gaxios@3.0.3: wanted: {"node":">=10"} (current: {"node":"8.17.0","npm":"6.13.4"})
npm WARN notsup Not compatible with your version of node/npm: gaxios@3.0.3
npm WARN notsup Unsupported engine for google-p12-pem@3.0.1: wanted: {"node":">=10"} (current: {"node":"8.17.0","npm":"6.13.4"})
npm WARN notsup Not compatible with your version of node/npm: google-p12-pem@3.0.1


+ firebase-tools@8.4.0
added 537 packages from 354 contributors and audited 538 packages in 23.509s

$ npm ls google-auth-library
└─┬ firebase-tools@8.4.0
  ├─┬ @google-cloud/pubsub@1.7.3
  │ └── google-auth-library@5.10.1  deduped
  ├── google-auth-library@5.10.1
  └─┬ google-gax@1.12.0
    └── google-auth-library@5.10.1  deduped
```

I can add a yarn resolution in the meantime, but thought I'd pass on the info for others in the same boat :)

**Comment by samtstern** [comment]
Looks like `grpc-js` added a dependency on `google-auth-library` 6+ in version `1.0.4`:
https://github.com/grpc/grpc-node/commit/010ef569f7f83007642f3f69e0ce13dee615d680

**Comment by samtstern** [comment]
@alexlouden reported this here:
https://github.com/grpc/grpc-node/issues/1442

And it is already being fixed here:
https://github.com/grpc/grpc-node/issues/1443

This is an issue with `grpc-js` and although it affects Firebase there's nothing we can do to fix it, we'll have to wait for their next patch release which should fix things up.

Thank you for reporting!

**Comment by samtstern** [comment]
The fix for this issue is included in `grpc-js` version `1.0.5` and higher.
